packages/stockflux-components/src/components/preview-window/PreviewWindow.js

```diff
@@ -0,0 +1,104 @@
+import React, { useState, useCallback, useEffect } from 'react';
+import { StockFluxHooks } from 'stockflux-core';
+
+export default ({
+  htmlfile = 'transparent-window.html',
+  children,
+  position = { left: 100, top: 100 },
+  size = { width: 300, height: 300 },
+  display
+}) => {
+  const TRANSPARENT_WINDOW_NAME = 'transparent_window';
+  const TRANSPARENT_CSS_PATCH = 'transparentWindow.css';
+
+  const windowStatus = {
+    show: 'SHOW',
+    showing: 'SHOWING',
+    hiding: 'HIDING',
+    hide: 'HIDE',
+    hidden: 'HIDDEN'
```

Comment:
The transitioning states were added to deal with the animation of the window fading in and out, as there were issues if you kept making the preview appear and disappear within its animation times.

For the example we are using on the watch list there is low chance of causing this now after it was changed to display on hover start, and not when you leave the window. But as this should be reusable in other areas, this will be required for where the transparent windows and display/hide rapidly and in succession.
The 'Show' status was completely unneeded though so its been removed.